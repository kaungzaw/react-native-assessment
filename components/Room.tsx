import { StyleSheet } from "react-native";
import { ThemedView } from "./ThemedView";
import { ThemedText } from "./ThemedText";
import { useThemeColor } from "@/hooks/useThemeColor";
import { RoomAvailability } from "@/hooks/useRooms";

type RoomProps = RoomAvailability & {};

export default function Room({ name, capacity, level, available }: RoomProps) {
  const customBackgroundColor = useThemeColor({}, "customBackground");

  const styles = StyleSheet.create({
    container: {
      paddingHorizontal: 20,
      paddingVertical: 10,
      backgroundColor: customBackgroundColor,
      borderRadius: 10,
      marginBottom: 10,
    },
    subContainer: {
      backgroundColor: "transparent",
      flexDirection: "row",
      justifyContent: "space-between",
    },
    box: {
      backgroundColor: "transparent",
      height: 2,
    },
    availableText: { color: "lightgreen", fontStyle: "italic" },
    notAvailableText: { color: "lightgrey", fontStyle: "italic" },
  });

  return (
    <ThemedView style={styles.container}>
      <ThemedView style={styles.subContainer}>
        <ThemedText type="defaultSemiBold">{name}</ThemedText>
        {available ? (
          <ThemedText style={styles.availableText}>Available</ThemedText>
        ) : (
          <ThemedText style={styles.notAvailableText}>Not Available</ThemedText>
        )}
      </ThemedView>
      <ThemedView style={styles.box} />
      <ThemedView style={styles.subContainer}>
        <ThemedText>Level {level}</ThemedText>
        <ThemedText>{capacity} Pax</ThemedText>
      </ThemedView>
      <ThemedView style={styles.box} />
    </ThemedView>
  );
}
