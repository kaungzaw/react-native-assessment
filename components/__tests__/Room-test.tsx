import * as React from "react";
import { render } from "@testing-library/react-native";

import Room from "../Room";

it(`renders correctly`, () => {
  const tree = render(
    <Room name="Room Name" capacity={2} level={3} available />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

describe("Room", () => {
  test("Room renders correctly", () => {
    const { getByText } = render(
      <Room name="Room Name" capacity={2} level={3} available />
    );

    getByText("Room Name");
    getByText("2 Pax");
    getByText("Level 3");
  });
});
