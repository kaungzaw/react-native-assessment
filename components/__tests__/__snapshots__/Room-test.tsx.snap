// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "backgroundColor": "#f6f6f6",
        "borderRadius": 10,
        "marginBottom": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Room Name
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "color": "lightgreen",
            "fontStyle": "italic",
          },
        ]
      }
    >
      Available
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "height": 2,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Level 
      3
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      2
       Pax
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "height": 2,
        },
      ]
    }
  />
</View>
`;
